lightningd/peer_htlcs.c

```diff
@@ -1602,6 +1584,25 @@ void peer_got_commitsig(struct channel *channel, const u8 *msg)
 				    tal_hex(channel, msg));
 		return;
 	}
+
+	/* If we're not synced with bitcoin network, we can't accept
+	 * any new HTLCs.  We stall at this point, in the hope that it
+	 * won't take long! */
+	if (added && !topology_synced(ld->topology)) {
```

Comment:
Is it possible then to process failed and fulfilled HTLCs?

It at least seems safe, unless the CLTV delta is such that the downstream (i.e., the failed) HTLC isn't fully settled (claimed) before the upstream timeout expires. In that case we'd fail the upstream HTLC but would still pay out the downstream one. Then again, if that were the case we would have chosen our parameters poorly.